language/move-prover/bytecode-to-boogie/src/env.rs

```diff
@@ -0,0 +1,726 @@
+        let field_view =
+            FieldDefinitionView::new(&self.data.module, self.data.module.field_def_at(*idx));
+        let struct_name = field_view.member_of().name();
+        self.find_struct(struct_name).expect("struct undefined")
+    }
+
+    /// Returns iterator over structs in this module.
+    pub fn get_structs(&'env self) -> impl Iterator<Item = StructEnv<'env>> {
+        self.data.struct_data.iter().map(move |data| StructEnv {
+            module_env: self,
+            data,
+        })
+    }
+
+    /// Globalizes a signature local to this module.
+    pub fn globalize_signature(&self, sig: &SignatureToken) -> GlobalType {
+        match sig {
+            SignatureToken::Bool => GlobalType::Bool,
+            SignatureToken::U8 => GlobalType::U8,
+            SignatureToken::U64 => GlobalType::U64,
+            SignatureToken::U128 => GlobalType::U128,
+            SignatureToken::ByteArray => GlobalType::ByteArray,
+            SignatureToken::Address => GlobalType::Address,
+            SignatureToken::Reference(t) => {
+                GlobalType::Reference({ Box::new(self.globalize_signature(&*t)) })
+            }
+            SignatureToken::MutableReference(t) => {
+                GlobalType::MutableReference({ Box::new(self.globalize_signature(&*t)) })
+            }
+            SignatureToken::TypeParameter(index) => GlobalType::TypeParameter(*index),
+            SignatureToken::Struct(handle_idx, args) => {
+                let struct_view = StructHandleView::new(
+                    &self.data.module,
+                    self.data.module.struct_handle_at(*handle_idx),
+                );
+                let declaring_module_env = self
+                    .env
+                    .find_module(&struct_view.module_id())
+                    .expect("undefined module");
+                let struct_env = declaring_module_env
+                    .find_struct(struct_view.name())
+                    .expect("undefined struct");
+                GlobalType::Struct(
+                    declaring_module_env.data.idx,
+                    struct_env.get_def_idx(),
+                    self.globalize_signatures(args),
+                )
+            }
+        }
+    }
+
+    /// Globalizes a list of signatures.
+    fn globalize_signatures(&self, sigs: &[SignatureToken]) -> Vec<GlobalType> {
+        sigs.iter()
+            .map(|s| self.globalize_signature(s))
+            .collect_vec()
+    }
+
+    /// Gets a list of type actuals associated with the index in the bytecode.
+    pub fn get_type_actuals(&self, idx: LocalsSignatureIndex) -> Vec<GlobalType> {
+        let actuals = &self.data.module.locals_signature_at(idx).0;
+        self.globalize_signatures(actuals)
+    }
+
+    /// Converts an address pool index for this module into a number representing the address.
+    pub fn get_address(&self, idx: &AddressPoolIndex) -> BigInt {
```

Comment:
The second part of this might fit well as as a method on `AccountAddress`. I imagine treating addresses as numbers would also be useful elsewhere.